src/pocketmine/entity/Villager.php

```diff
@@ -50,11 +60,17 @@ protected function initEntity(){
 		}
 
 		$this->setProfession($profession);
+		$this->setTraderName($this->namedtag->getString("TraderName", ""));
+		$this->setCanTrading((bool) $this->namedtag->getByte("CanTrading", 0));
+		$this->setRecipes($this->namedtag->getListTag(TradeRecipe::TAG_RECIPES) ?? new ListTag(TradeRecipe::TAG_RECIPES, []));
 	}
 
 	public function saveNBT(){
 		parent::saveNBT();
 		$this->namedtag->setInt("Profession", $this->getProfession());
+		$this->namedtag->setByte("CanTrading", (int) $this->isCanTrading());
+		$this->namedtag->setString("TraderName", $this->getTraderName());
```

Comment:
`TraderName`? What? What is a trader? The trading gui should be set to the name of the villager entity.